src/vcpkg/commands.generate-message-map.cpp

```diff
@@ -215,31 +212,55 @@ namespace vcpkg::Commands
         }
         std::sort(messages.begin(), messages.end(), MessageSorter{});
 
-        bool has_incorrect_comment = false;
+        bool has_errors = false;
         LocalizedString format_string_parsing_error;
         Json::Object obj;
         for (Message& msg : messages)
         {
+            if (msg.name != "ErrorMessage" && Strings::case_insensitive_ascii_starts_with(msg.value, "error:"))
+            {
+                has_errors = true;
+                msg::print_error(msgErrorMessageMustUsePrintError, msg::value = msg.name);
+            }
+
+            if (msg.name != "WarningMessage" && Strings::case_insensitive_ascii_starts_with(msg.value, "warning:"))
+            {
+                has_errors = true;
+                msg::print_error(msgWarningMessageMustUsePrintWarning, msg::value = msg.name);
+            }
+
+            if (Strings::contains(msg.value, "   "))
+            {
+                has_errors = true;
+                msg::print_error(msgLocalizedMessageMustNotContainIndents, msg::value = msg.name);
+            }
+
+            if (!msg.value.empty() && msg.value.back() == '\n')
+            {
+                has_errors = true;
+                msg::print_error(msgLocalizedMessageMustNotEndWithNewline, msg::value = msg.name);
+            }
+
             auto mismatches = get_format_arg_mismatches(msg.value, msg.comment, format_string_parsing_error);
             if (!format_string_parsing_error.data().empty())
             {
-                msg::println(msgGenerateMsgErrorParsingFormatArgs, msg::value = msg.name);
-                Checks::msg_exit_with_message(VCPKG_LINE_INFO, format_string_parsing_error);
+                has_errors = true;
+                msg::print_error(msg::format(msgGenerateMsgErrorParsingFormatArgs, msg::value = msg.name)
+                                     .append(format_string_parsing_error));
             }
 
             if (!mismatches.arguments_without_comment.empty() || !mismatches.comments_without_argument.empty())
             {
-                has_incorrect_comment = true;
-                msg::print(comments_msg_color, comments_msg_type);
-                msg::println(comments_msg_color, msgGenerateMsgIncorrectComment, msg::value = msg.name);
+                has_errors = true;
+                msg::print_error(msgGenerateMsgIncorrectComment, msg::value = msg.name);
 
                 for (const auto& arg : mismatches.arguments_without_comment)
                 {
-                    msg::println(comments_msg_color, msgGenerateMsgNoCommentValue, msg::value = arg);
+                    msg::println(Color::error, msgGenerateMsgNoCommentValue, msg::value = arg);
```

Comment:
```suggestion
                    msg::print_error(msgGenerateMsgNoCommentValue, msg::value = arg);
```
?

Reply:
That "error:" was already printed by 255